lib/apiservers/engine/backends/archive.go

```diff
@@ -190,9 +198,18 @@ func (c *Container) ContainerStatPath(name string, path string) (stat *types.Con
 		return nil, NotFoundError(name)
 	}
 
+	path = filepath.Clean(path)
 	mounts := mountsFromContainer(vc)
 	mounts = append(mounts, types.MountPoint{Destination: "/"})
 
+	// handle the special case of targeting a volume mount point before it exists.
+	// this will be important for non started container cp, will also be important
+	// to certain behaviors for diff on a non started container.
+	if stat, succeed := tryFakeStatPath(mounts, path); succeed {
```

Comment:
I think an additional detail for this case is as such: This is the personality so it will be a docker specific ignore. The other cases are that this path is a file or such type situation. Since this is an actual path of a disk mount on the container and the personality knows it, the personality can reasonably assume that the container mount path is valid since a `docker create` must successfully execute before this(meaning the mount point is valid). That being said we should note that currently VIC has a bug where we do not confirm if the mount point is valid on a `create` operation. Leading us to create broken containers that can never start due to the mount points actually being invalid targets. Since we have a useful `StatPath` from the portlayer we can now fix that during a `docker create` and this code will make more sense. We might still make a case for attempting the actual stat results as it could help us understand if we attempted to stat the wrong thing or some other such scenario. That is my $0.02.